### PR TITLE
Backporting Virgo-style loot piles from Lighthouse.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Ports Polaris-style loot piles from Lighthouse: https://github.com/OpaqueSunset/LighthouseRelay/tree/master/mods/content/dungeon_loot
- Adds a few to the tradeship for testing purposes.

## Why and what will this PR improve
These are very good for random loot generation in away sites and will allow me to merge some bespoke junk pile handling in from Scav.

## Authorship
~~I believe Penny~~ ProbablyCarl did the Lighthouse port from Virgo/Polaris/CitRP but I'm not sure who the original coder was.

## Changelog
:cl:
tweak: Loot piles have been added to the Tradeship underside.
/:cl:

### CODE REVIEW

File: mods/content/dungeon_loot/loot_parent.dm [hunk 1]
```diff
@@ -0,0 +1,110 @@
+/obj/structure/loot_pile
+	abstract_type = /obj/structure/loot_pile
+	name = "base loot pile"
+	desc = "If you can read me, this is bugged."
+	icon = 'mods/content/dungeon_loot/icons/obj/loot_piles.dmi'
+	icon_state = "randompile"
+	density = FALSE
+	anchored = TRUE
+
+	var/list/icon_states_to_use = list() // List of icon states the pile can choose from on initialization. If empty or null, it will stay the initial icon_state.
```
Comment (MistakeNot4892) [inline mods/content/dungeon_loot/loot_parent.dm]: Should probably be a static list returned to a proc a la random spawners.

File: mods/content/dungeon_loot/loot_parent.dm [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+	var/chance_nothing = 0			// Unlucky people might need to loot multiple spots to find things.
+
+	var/chance_uncommon = 10		// Probability of pulling from the uncommon_loot list.
+	var/chance_rare = 1				// Ditto, but for rare_loot list.
+	var/loot_depletion = FALSE		// If true, loot piles can be 'depleted' after a certain number of searches by different players, where no more loot can be obtained.
+	var/loot_left = 0				// When this reaches zero, and loot_depleted is true, you can't obtain anymore loot.
+	var/delete_on_depletion = FALSE	// If true, and if the loot gets depleted as above, the pile is deleted.
+
+	var/list/common_loot = list()	// Common is generally less-than-useful junk and filler, at least for maint loot piles.
```
Comment (MistakeNot4892) [inline mods/content/dungeon_loot/loot_parent.dm]: As above, should probably be static proc-returned lists.

File: mods/content/dungeon_loot/loot_parent.dm [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	var/list/uncommon_loot = list()	// Uncommon is actually maybe some useful items, usually the reason someone bothers looking inside.
+	var/list/rare_loot = list()		// Rare is really powerful, or at least unique items.
+
+/obj/structure/loot_pile/attack_ai(var/mob/user)
+	if(isrobot(user) && Adjacent(user))
+		return attack_hand(user)
```
Comment (MistakeNot4892) [inline mods/content/dungeon_loot/loot_parent.dm]: Probably needs a `return ..()`

**Comment by MistakeNot4892** [comment]
I've removed the alien ones as we have no items for that on main neb, and I've converted the mecha ones to our premade exosuits. Also populated the medical ones. Items I chose are completely arbitrary, open to feedback on them.

**Comment by out-of-phaze** [comment]
For attribution purposes, the port was done by @ProbablyCarl, not by me.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**